### ISSUE TITLE
amphibious breakage too easy - how to roll from land into water or visa versa

### ISSUE THREAD

**Issue by wkitty42** [Juanvvc/c172p-detailed#307]
trying to go from land to water or water to land is currently very painful... you have to keep raising and lowering the gear to ""bounce/walk" the craft in to or out of the water unless you get a really fast run and can force a slide across the water/land transition boundary... doing the gear "bounce/walk" too rapidly much too easily bends the pontoon and/or breaks the gear...

the too easy breakage is one part of this issue...
the inability to roll in to or out of the water is the other part of this issue...

i did a water landing last evening and had to keep my speed up when approaching the land so i would slide up onto it pretty decently (throwing sparks) and then i could lower the gear and roll on over to a parking spot about 3 feet from and facing the water... this morning, i started the craft and tried to roll into the water for a hopefully good water takeoff... i finally just quit the sim when i couldn't get the craft into the water as there's no way to back up either... it isn't like we can reverse the prop pitch ;)

**Comment by dany93** [comment]
Does a slope exist in the FG Terrain to go from land to water and vice versa? I doubt it. IRL, trying this on a step breaks certainly... Like raising / lowering the gear to "jump". Nobody would do this IRL.

**Comment by wlbragg** [comment]
Yes, I have played with this also and was disappointing by the outcome. Hopefully we can fix this somehow. 
Interesting though about the ease of breaking, because it isn't easy to break on a hard landing or nose dive. I think the contact points that are used for damage conditions are misplaced.

**Comment by wkitty42** [comment]
on the easy breakage thing, try this... open the craft options and slide it off to your upper right... then go to heli view and slide around to the side of the craft... now, switch between normal gear, pontoons and amphibious... just click this one, and then randomly that one pretty quickly... the craft will hop all around and i've even had it do a flip :laughing: 

yesterday, i could easily break the gear or pontoons or amphibious by doing this... part of this has to do with the height differences... i even had the craft start rocking back and forth yesterday like a rocking horse kids toy... i couldn't stop it and it wouldn't settle down for some minutes... finally i started the engine, went to full power and as soon as i started rolling, it settled down... then i went on to fly the above water landing, camping and attempted morning fly out...

i haven't been able to make this rocking or breakage happen again but all of this may also be an anomaly that i see when i go look at the original c172p and then come back to this -detailed version...

@dany93 yes, i know that IRL one wouldn't do this... they'd just get out and push the craft but we can't do that (yet)... i guess that would maybe be a feature for the walker but we're not to that point yet :wink:

**Comment by wkitty42** [comment]
the main point i guess i'm trying to make is that just like in writing programs, we have to protect from the random crazy stuff that users will throw at the program at some point... we used to find all sorts of security holes in POTS dial up BBS software by just hitting random keys at their menus...

**Comment by dany93** [comment]
In FG, I think that the boundary between ground and water is just a line at the surface. Nothing exists below, thus my guess about the non-existence of a slope which would extend the ground under water.

**Comment by wkitty42** [comment]
@dany93 right... what i was hoping for was that the amphibious wheels could still be extended when entering the water so that the front of the pontoon would start floating... then when the rest of the craft was in the water and floating completely on the pontoons, then the gear would be raised... part of the problem, in the case i described above, seemed to be soft ground as the front wheels sank into the ground maybe a foot before the water line... previous to that, they were on hard ground... that's why i have decided to try for running fast toward the water and then raising the gear so as to slide (and spark) on the ground right into the water... kind of like sliding into base in baseball... just need to get the speed and gear raising timing right until something else can be done (like maybe relaxing the hydroFDM when the wheels are down and we're at say 10 kts or less)...

note that i do not have my 'transition' setting in the advanced rendering turned up because it drops my fps to 1 at best...

**Comment by andgi** [comment]
Getting a water-ground or ground-water transition to work in FG is tricky.
Once you leave solid ground you are in infinitely deep water so as you approach the shore one or two of your landing gears just sink. Then you are either standing on one or both of the step contact points (if they are over solid ground) - and they have too high friction set. A similar problem occurs when going from water to solid ground.

Adding a slipway scenery object that extends under the water surface would allow trying a more realistic transition. I expect few people launch their floatplane by pushing it off a vertical wharf... and if they do they have the benefit of having floats with a continuous solid keel while we in FG have a void with 3 contact points...

**Comment by andgi** [comment]
Even with a ramp it is not so easy to get out of the water...
http://www.gidenstam.org/users/anders/FlightGear/screenshots/fgfs-c172p-321.jpg

Apparently (but not surprisingly) the ground intersection code seems to stop with the first hit (solid or not) and doesn't look further - so it doesn't appear to see the solid ramp under the water surface.

**Comment by wkitty42** [comment]
sounds like a feature for FGFS 3.8 :wink: :wink: 
it may open the doors for more floatplanes in FGFS and even offer more opportunities for vehicles/craft (eg: a car or truck pulling a boat and backing it into the water on a boating ramp) :smile_cat:

**Comment by andgi** [comment]
If you want to try the slipway/ramp in the picture above, it will appear in terrasync scenery shortly (I think ~noon UTC).
To start on the slipway use --lat=57.70266 --lon=11.78678 --heading=135. The closest airport is ESGP but the slipway is hard to spot from the air.

**Comment by wkitty42** [comment]
i will try to take a look at this... i'm currently scrambling to get today's tasks all done in a few hours... we lost our modem this morning ~0500 and weren't able to get a replacement and get back online until ~1500...

**Comment by gilbertohasnofb** [comment]
Is this issue still relevant or can I close it? If still relevant then I will add labels and a milestone (3.8) to it.

**Comment by andgi** [comment]
I'd say the beaching aspect is still relevant but will probably need some adjustment in JSBSim's ground callback code to work. I have not yet checked if it is possible to specifically ask the ground cache object in FlightGear about solid/non-solid surfaces so that a solid surface below a non-solid can be detected.

**Comment by wkitty42** [comment]
the labels and milestone tags are good for this... i still have my challenge on the forums but not many have attempted it or else they just haven't posted about it... i'm unable to try any more until the throttle keys are returned to their previous state... at that time i had more control over my throttle for landing speed but that's all changed since then :(

**Comment by legoboyvdlp** [comment]
@dany93 
There is a 3d model of a slope somewhere in Scandinavia...
Check what/where did you fly today, about 10 pages ago.

**Comment by andgi** [comment]
@dany93 @legoboyvdlp The ESGB slipway is here: --lat=57.70266 --lon=11.78678 --heading=135

**Comment by gilbertohasnofb** [comment]
Is this still relevant or can I close it? I think our float and amphibious FDM's are very nice now.

**Comment by andgi** [comment]
@gilbertohasnofb I think the problems with ground-water and water-ground transitions described here still remain. I have not yet have the time to look into the ground cache in FG and its JSBSim adapter to see if one can query for a solid surface below a non-solid one.

An additional heads up: I spent some time during the summer with modelling planing hydrodynamics from (kind of) first principles. I have not yet applied this to the c172p but my MFI-9B testbed (a much less finished aircraft than the c172p, though) is available here: http://github.com/andgi/FlightGear-MFI-9

**Comment by gilbertohasnofb** [comment]
@andgi :+1: we leave this opened then!

**Comment by wlbragg** [comment]
Part of this issue is fixed. That is the landing on floats on a hard surface without damaging them.
https://www.dropbox.com/s/enrm3ujh0rhqah7/2018-04-04%2019-07-29.mkv?dl=1
Sorry for the quality. I need to reboot and free up some resources but don't want to take the time.
Let to do is the transition from water to land.
